charmstore/lib.py

```diff
@@ -115,10 +109,18 @@ def __init__(self, id=None, api=None, timeout=None):
             )
 
     def revisions(self):
-        data = self.raw.get('revision-info', {}).get('Revisions', [])
+        if self.revisions is None:
```

Comment:
As mentioned above, there's a conflict with this and the method name.  Something like `self._revisions` should be used as the caching variable instead.

Reply:
These references still need to be fixed.